### PR TITLE
added /verbose command, refactored bolt_app.py and utils.py

### LINKED ISSUE #61: Create a `/verbose` slash command in the slackbot to show the thought process
I notice people rarely use the `-verbose` flag. Sometimes I just want to see the thought process after the query, and I don't want to run the query again just to include -verbose. Having a function that reveals the thought process after the query reduces that friction.

I found a tutorial for creating slash commands, it seems simple enough.
https://www.youtube.com/watch?v=sEFx0b9y_Xo&list=PLzMcBGfZo4-kqyzTzJWCV6lyK-ZMYECDc&index=3

### PR DESCRIPTION
**Pull request by johnnykfeng**
**List of changes**
- refactored `get_response()` to return always return response and logger
- formatting logger to verbose_message done inside "app_mention" endpoint
- added global variable `stored_verbose_message`
- added /verbose command to display `stored_verbose_message`
- exported helper functions to `utils.py`

@saminegash Please add /verbose slash command in the slackbot settings
Follow this video https://youtu.be/sEFx0b9y_Xo?t=59

Resolves #61

### CODE REVIEW

File: apps/slackbot/bolt_app.py
Comment (saminegash) [inline apps/slackbot/bolt_app.py:81]: can we maybe kind of add support for post here?

File: apps/slackbot/utils.py
Comment (saminegash) [inline apps/slackbot/utils.py:26]: why are we checking for this?

File: apps/slackbot/utils.py
Comment (saminegash) [inline apps/slackbot/utils.py:36]: remove this

File: apps/slackbot/utils.py
Comment (saminegash) [inline apps/slackbot/utils.py:60]: remove block of comments

**Review by saminegash — CHANGES_REQUESTED** [review]

**Comment by saminegash** [comment]
Thanks for the PR; please check these comments.

File: apps/slackbot/utils.py
Comment (johnnykfeng) [reply apps/slackbot/utils.py:26]: the idea is to have -verbosex give a different response than -verbose

**Comment by 20001LastOrder** [comment]
> I tried to remove the comments in utils.py but it created another PR. Maybe close this PR and just review/merge PR#76... Sorry for making this more complicated than it is

You can merge your changes to the [johnnykfeng:main](https://github.com/johnnykfeng/practical-llms/tree/main) branch and then the PR will be updated automatically

File: apps/slackbot/bolt_app.py
Comment (johnnykfeng) [reply apps/slackbot/bolt_app.py:81]: @saminegash Actually I might be wrong here. Don't merge this yet... Let me do some more test.

**Comment by johnnykfeng** [comment]
Alright I figured it out. Leave the code in `bolt_app.py` as it is. Enter the flask app backend URL in Request URL box with `/slack/events` added to the end. In my environment, it works with the flask app deployed with ngrok.
However, this feature is not needed right now, so maybe wait until next week to merge it

image

**Comment by saminegash** [comment]
still you need to remove the comments resolve the comments I have given you!

File: apps/slackbot/utils.py
Comment (johnnykfeng) [reply apps/slackbot/utils.py]: see third commit in this PR

File: apps/slackbot/utils.py
Comment (johnnykfeng) [reply apps/slackbot/utils.py]: see third commit in this PR

**Comment by johnnykfeng** [comment]
I just closed this PR because I want to make a few more changes to this feature. Instead of changing this PR, I will make an entirely new PR for cleaner merge

**Pull request closed without merging**
